### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by GRSARVESH** [codeclassifiers/loginforms#15] [Q&A] [answered]

**Comment by saurabhnative** [comment, marked as answer]
Hello @GRSARVESH,
The page will not switch because we need to set up a backend server first from which an access token is requested for redirection. Please check the code from line no: 28 in the below file:-
https://github.com/codeclassifiers/loginforms/blob/master/src/components/RegistrationForm/RegistrationForm.js

If you need to understand the process of setting up the backend server, please check the complete tutorial here:-
https://dev.to/dipakkr/implementing-authentication-in-nodejs-with-express-and-jwt-codelab-1-j5i

which is also mentioned in my own article for using this repository:-
https://medium.com/technoetics/create-basic-login-forms-using-react-js-hooks-and-bootstrap-2ae36c15e551

**Comment by GRSARVESH** [comment]


**Reply by saurabhnative** [reply]
Refer to this docs on how to setup environment variables in react.js projects:-
https://create-react-app.dev/docs/adding-custom-environment-variables/#adding-development-environment-variables-in-env

I am using a similar environment variable in apiConstants.js file:-
https://github.com/codeclassifiers/loginforms/blob/master/src/constants/apiConstants.js

Assign an entry for `REACT_APP_SERVER_URL` in your .env file with the base url of your backend application.
In case of any issues check the browser and node.js terminal consoles and debug the errors accordingly.

**Comment by saurabhnative** [comment]
Hello @GRSARVESH,
You can find reference code for backend routes in this file:-
https://github.com/codeclassifiers/quiz-app-backend/blob/master/routes/users.js

It's a different application which uses login/registration modules from this project.
